--- librime-1.7.3/src/rime/gear/table_translator.cc	2021-02-11 15:17:48.000000000 +0000
+++ librime-1.7.3-new/src/rime/gear/table_translator.cc	2021-12-17 05:17:52.889104833 +0000
@@ -7,6 +7,7 @@
 #include <boost/algorithm/string.hpp>
 #include <boost/range/adaptor/reversed.hpp>
 #include <utf8.h>
+#include <math.h>
 #include <rime/candidate.h>
 #include <rime/common.h>
 #include <rime/composition.h>
--- librime-1.7.3/src/rime/gear/script_translator.cc	2021-02-11 15:17:48.000000000 +0000
+++ librime-1.7.3-new/src/rime/gear/script_translator.cc	2021-12-17 05:18:06.272276933 +0000
@@ -10,6 +10,7 @@
 #include <stack>
 #include <boost/algorithm/string/join.hpp>
 #include <boost/range/adaptor/reversed.hpp>
+#include <math.h>
 #include <rime/composition.h>
 #include <rime/candidate.h>
 #include <rime/config.h>
