diff -Naur uniconvertor-2.0rc5/setup.cfg.in uniconvertor-2.0rc5.mod/setup.cfg.in
--- uniconvertor-2.0rc5/setup.cfg.in	2022-03-03 18:11:49.714395548 -0600
+++ uniconvertor-2.0rc5.mod/setup.cfg.in	2022-03-03 18:09:47.940438321 -0600
@@ -1,12 +1,3 @@
-[install]
-install-base=$APP_INSTALL_PATH
-install-lib=$APP_INSTALL_PATH
-install-purelib=$APP_INSTALL_PATH
-install-platlib=$APP_INSTALL_PATH
-install-scripts=/usr/bin
-install-data=$APP_INSTALL_PATH
-install-headers=$APP_INSTALL_PATH
-
 [bdist_rpm]
 release = 1
 packager = Ihor Novikov <sk1.project.org@gmail.com>
