From b65de5ddabadd1c4b6d8e76d7215dcd1006c02ca Mon Sep 17 00:00:00 2001
From: Tianhao Chai <cth451@gmail.com>
Date: Sat, 21 May 2022 15:56:26 -0500
Subject: [PATCH] udma-barrier: provide memory barrier on loongson3

---
 util/udma_barrier.h | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/util/udma_barrier.h b/util/udma_barrier.h
index 593d5b39..78cdb119 100644
--- a/util/udma_barrier.h
+++ b/util/udma_barrier.h
@@ -98,6 +98,8 @@
 #define udma_to_device_barrier() asm volatile("" ::: "memory")
 #elif defined(__loongarch__)
 #define udma_to_device_barrier() asm volatile("dbar 0" ::: "memory")
+#elif defined(__MIPSEL__) && defined(__mips64)
+#define udma_to_device_barrier() asm volatile("sync 0" ::: "memory")
 #elif defined(__riscv)
 #define udma_to_device_barrier() asm volatile("fence ow,ow" ::: "memory")
 #else
@@ -134,6 +136,8 @@
 #define udma_from_device_barrier() asm volatile("" ::: "memory")
 #elif defined(__loongarch__)
 #define udma_from_device_barrier() asm volatile("dbar 0" ::: "memory")
+#elif defined(__MIPSEL__) && defined(__mips64)
+#define udma_from_device_barrier() asm volatile("sync 0" ::: "memory")
 #elif defined(__riscv)
 #define udma_from_device_barrier() asm volatile("fence ir,ir" ::: "memory")
 #else
@@ -202,6 +206,8 @@
 #define mmio_flush_writes() asm volatile("" ::: "memory")
 #elif defined(__loongarch__)
 #define mmio_flush_writes() asm volatile("dbar 0" ::: "memory")
+#elif defined(__MIPSEL__) && defined(__mips64)
+#define mmio_flush_writes() asm volatile("sync 0" ::: "memory")
 #elif defined(__riscv)
 #define mmio_flush_writes() asm volatile("fence ow,ow" ::: "memory")
 #elif defined(__s390x__)
-- 
2.36.0

